Content.Server/GameObjects/Components/Chemistry/InjectorComponent.cs

```diff
@@ -0,0 +1,240 @@
+﻿using System;
+using Content.Server.GameObjects.Components.Metabolism;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Server.Interfaces;
+using Content.Shared.Chemistry;
+using Content.Shared.GameObjects.Components.Chemistry;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Chemistry
+{
+    /// <summary>
+    /// Server behavior for reagent injectors and syringes. Can optionally support both
+    /// injection and drawing or just injection. Can inject/draw reagents from solution
+    /// containers, and can directly inject into a mobs bloodstream.
+    /// </summary>
+    [RegisterComponent]
+    public class InjectorComponent : SharedInjectorComponent, IAfterAttack, IUse
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IServerNotifyManager _notifyManager;
+        [Dependency] private readonly ILocalizationManager _localizationManager;
```

Comment:
You can use the static `Loc` class instead of pulling an IoC dependency on `ILocalizationManager`, so you know.

Reply:
Changed